tekton/ci-workspace/jobs/tekton-image-build.yaml

```
@@ -0,0 +1,152 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kaniko4ci
+spec:
+  description: >-
+    This Task builds source into a container image using Google's kaniko tool.
+
+    Copied from https://hub.tekton.dev/tekton/task/kaniko.
+    It allows defaulting the IMAGE parameter from the context
+
+  params:
+  - name: DOCKERFILE
+    description: Path to the Dockerfile to build.
+    default: ./Dockerfile
+  - name: CONTEXT
+    description: The build context used by Kaniko.
+    default: ./
+  - name: EXTRA_ARGS
+    default: ""
+  - name: pullRequestNumber
+    description: pull request number, used for the tag name
+  workspaces:
+  - name: source
+  results:
+  - name: IMAGE-DIGEST
+    description: Digest of the image just built.
+
+  steps:
+  - name: build-and-push
+    workingDir: $(workspaces.source.path)
+    image: gcr.io/kaniko-project/executor@sha256:e00dfdd4a44097867c8ef671e5a7f3e31d94bd09406dbdfba8a13a63fc6b8060  # debug image
+    # specifying DOCKER_CONFIG is required to allow kaniko to detect docker credential
+    # https://github.com/tektoncd/pipeline/pull/706
+    env:
+    - name: DOCKER_CONFIG
+      value: /tekton/home/.docker
+    script: |
+      # Setup the image name
+      CONTEXT=$(workspaces.source.path)/$(params.CONTEXT)
+      IMAGE_NAME=$(basename "${CONTEXT}"):pr-$(params.pullRequestNumber)
+      # Run the build
+      /kaniko/executor \
+        - $(params.EXTRA_ARGS) \
+        --dockerfile=$(params.DOCKERFILE) \
+        --context=$(workspaces.source.path)/$(params.CONTEXT)  # The user does not need to care the workspace and the source.
+        --destination=${IMAGE_NAME}
+        --digest-file=$(results.IMAGE-DIGEST.path)
+      # kaniko assumes it is running as root, which means this example fails on platforms
+      # that default to run containers as random uid (like OpenShift). Adding this securityContext
+      # makes it explicit that it needs to run as root.
+    securityContext:
+      runAsUser: 0
+---
+apiVersion: custom.tekton.dev/v1alpha1
+kind: TaskLoop
+metadata:
+  name: image-build
+spec:
+  iterateParam: context
+  taskRef:
+    name: kaniko4ci
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: tekton-image-build
+  annotations:
+    description: |
+      Build the images of for all the Dockerfile modified by the PR
```

```suggestion
      Build the images for all the Dockerfiles modified by the PR
```
Maybe build the images of all the Dockerfiles modified by the PR. Of or For

---

Fixed